### ISSUE TITLE
incorrect-ident-case is inconsisten with rustc equivalents and provides wrong fixes

### ISSUE THREAD

**Issue by Maxpxt** [rust-lang/rust-analyzer#10473]
[incorrect-ident-case](https://rust-analyzer.github.io/manual.html#incorrect-ident-case) is sometimes inconsistent with rustc's non_upper_case_globals and non_snake_case and suggests weird fixes.

Examples:

```rust
// Triggers `rustc(non_upper_case_globals)`, which suggests `FOO_BAR` as a fix.
// Triggers `rust-analyzer(incorrect-ident-case)`, which suggests `F_O_O_B_AR` as a fix.
pub const FOO_BAr: u32 = 42;

// Does not trigger any rustc error or warning.
// Triggers `rust-analyzer(incorrect-ident-case)`, which suggests `F_O_O__B_A_R` as a fix.
pub const FOO__BAR: u32 = 42;

#[allow(unused_variables)]
pub fn foo() {
    // Triggers `rustc(non_snake_case)`, which suggests `foo_bar` as a fix.
    // Triggers `rust-analyzer(incorrect-ident-case)`, which suggests `foo__bar` as a fix.
    let foo__bar = ();
}
```

rust-analyzer version: 5ff992465 2021-10-06 nightly

**Comment by Crauzer** [comment]
For the uppercase example, this is being caused by https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/stdx/src/lib.rs#L40
